### PR TITLE
[docs] Update dynamic partitions sensor examples to use `SensorResult`

### PR DESCRIPTION
**Pull request by clairelin135**
Change docs references instead to use `SensorResult` instead of `context.instance.add_dynamic_partitions(...)`

### CODE REVIEW

File: python_modules/dagster/dagster/_core/definitions/partition.py [hunk 1]
```diff
@@ -607,8 +607,10 @@ class DynamicPartitionsDefinition(
 
             @sensor(job=my_job)
             def my_sensor(context):
-                context.instance.add_dynamic_partitions(fruits.name, [partition_key])
-                return RunRequest(partition_key=partition_key)
+                return SensorResult(
+                    run_requests=[RunRequest(partition_key="apple")],
+                    dynamic_partitions_requests=[fruits.build_add_request([partition_key])]
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/partition.py:612]: Where is `partition_key` defined?

**Review by sryza — COMMENTED** [review]
Do `SensorResult` and the new request classes show up in the api doc?

File: python_modules/dagster/dagster/_core/definitions/partition.py (hunk 1 above)
Comment (clairelin135) [reply python_modules/dagster/dagster/_core/definitions/partition.py]: good catch, i've changed this

**Comment by clairelin135** [comment]
`SensorResult` and the request classes now show up in the sensor api docs

**Review by sryza — APPROVED** [review]

**Pull request merged**